code/datums/components/crafting/crafting.dm

```diff
@@ -205,13 +212,16 @@
 		return ", invalid recipe!" // This can happen, I can't really explain why, but it can. Better safe than sorry.
 
 	var/list/contents = get_surroundings(crafter, recipe.blacklist)
-	var/send_feedback = 1
-	var/turf/dest_turf = get_turf(crafter)
 
 	if(!check_contents(crafter, recipe, contents))
 		return ", missing component."
 
-	if(!check_tools(crafter, recipe, contents))
+	var/turf/dest_turf = get_turf(crafter)
+
+	//Tools are the most fundamental aspect of crafting after the components
+	//However this call is redundant during unit tests. There's another one right before we go on
+	//and spawn the object, and that's the only one we need to call anyway.
+	if(!PERFORM_ALL_TESTS(crafting) && !check_tools(crafter, recipe, contents))
```

Comment:
Don't exclude stuff from tests because its redundant unless it brings some degree of RNG/unsolvable incompatibility with testing environments